api/autocomplete.js

```diff
@@ -0,0 +1,13 @@
+const autocomplete = async (addressInput) => {
+  const result = await fetch(
+    "http://localhost:8000/api/autocomplete/location",
+    {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ query: addressInput }),
+    }
+  ).then((res) => res.json());
```

Comment:
First of all, lets rename this function to `autocompleteLocation` since in the future we will also need to add an `autocompleteSearch` to this same file. Also let's document what the function is for and add  `@param` and `@return` docs to see the types of data that is expected.

Additionally, If you do `.then` here, you can't grantee the data will always be accessible because the function is asynchronous when you try to access it. I recall you said that this was slower compared to the previous implementation. It's likely that the reason it's slower is because you are not getting the responses when expected. 

Instead you need to await both the request and the json data as both are asynchronous operations.

```javascript
const autocomplete = async (addressInput) => {
  const result = await fetch(
    "http://localhost:8000/api/autocomplete/location",
    {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ query: addressInput }),
    }
  );
  return await result.json(); // note how I await the json data
};
```

if you really want to do it together, like you did with the then statement, then you can add two awaits with parenthesis to mark where the await statements happen. Note that too many nested await statements can sometimes be hard to read, but here it is for reference anyways,

```js
const autocomplete = async (addressInput) => {
  return await (
    await fetch("http://localhost:8000/api/autocomplete/location", {
      method: "POST",
      headers: { "Content-Type": "application/json" },
      body: JSON.stringify({ query: addressInput }),
    })
  ).json();
};
```

Also just for best practices, if you at some point think there is a reason some of these APIs will break, don't forget to add try/catch blocks. In this case it might not be necessary, but its good for API calls that could break something. 

The way you are doing it now, you are calling .json() on a `Promise` object since you have `await` before the `fetch()` call.

Reply:
Ohhh I see what you saying!! I didn't think of that. My Bad.